planner/optimize.go

```diff
@@ -203,6 +208,40 @@ func handleInvalidBindRecord(ctx context.Context, sctx sessionctx.Context, stmtN
 	}
 }
 
+// IsPointGetWithoutDoubleRead returns true when meets following conditions:
+//  1. ctx is auto commit tagged
+//  2. plan is point get by pk, or point get by unique index (no double read)
+func IsPointGetWithoutDoubleRead(ctx sessionctx.Context, p plannercore.Plan) bool {
+	// check auto commit
+	if !ctx.GetSessionVars().IsAutocommit() {
+		return false
+	}
+
+	// check plan
+	if proj, ok := p.(*plannercore.PhysicalProjection); ok {
+		if len(proj.Children()) != 1 {
+			return false
+		}
+		p = proj.Children()[0]
+	}
+
+	switch v := p.(type) {
```

Comment:
I think we can simply check
```
_, ok := p.(*plannercore.PointGetPlan)
return ok
```

Reply:
Great, but should make sure `p.IndexInfo == nil` to avoid double read？